### PR TITLE
Feature push button widget

### PR DESCRIPTION
**Pull request by livecodegeorgia**
New push button widget with iOS and Android themes. The Android(Raised) theme is incomplete as it does not have a drop shadow.

### CODE REVIEW

File: extensions/widgets/pushButton/pushButton.lcb [hunk 1]
```diff
@@ -0,0 +1,504 @@
...
+Syntax: set the <backgroundOpacity> of <widget> to <pOpacity>
+Syntax: get the <backgroundOpacity> of <widget>
+
+Summary: The background opacity of the button
+
+Parameters:
+pOpacity(enum): The background opacity of the button
+-"Opaque"
+-"Translucent"
+
```
Comment (livecodeali) [inline extensions/widgets/pushButton/pushButton.lcb]: Transparent is missing from this list.

File: extensions/widgets/pushButton/pushButton.lcb [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+   put "iOS" into mWidgetTheme
+   put "Button" into mButtonLabel
+   put color [0, 121/255, 1] into mTextColor
+   put color [1, 1, 1] into mBackgroundColor
+   put "Transparent" into mBackgroundOpacity
+
+   put false into mShowBorder
+   put color [0, 121/255, 1] into mDefaultColor
+   put (my height)*(17/29) into mIosFontSize
+   put (my height)*(14/36) into mAndroidFontSize
```
Comment (livecodeali) [inline extensions/widgets/pushButton/pushButton.lcb]: you could just call updateVariables() here

File: extensions/widgets/pushButton/pushButton.lcb [hunk 3]
```diff
@@ -0,0 +1,504 @@
...
+
+   set the font of this canvas to fetchFont()
+   set the paint of this canvas to fetchPaint("label")
+   fill text tLabel at center of my bounds on this canvas
+end handler
+
+private handler updateVariables() returns nothing
+   put (my height)*(17/29) into mIosFontSize
+   put (my height)*(14/36) into mAndroidFontSize
+end handler
```
Comment (livecodeali) [inline extensions/widgets/pushButton/pushButton.lcb]: Can you make these proportions constants, so it's obvious what the numbers represent and makes it easier to change

**Comment by livecodeali** [comment]
Can you add a release note for this too please? Here's an example of the one for when the JSON library was added:
https://github.com/livecode/livecode/commit/f3db2aea52853b4e1c2c58d2149f8aa48af58812

File: extensions/widgets/pushButton/pushButton.lcb (hunk 3 above)
Comment (livecodegeorgia) [inline extensions/widgets/pushButton/pushButton.lcb]: I did think about doing that, but I cannot set a constant to 17/29 and 17/29 is 0.5862068965517241379310344827586206896551724137931034... (similar issue for 14/36) 

Maybe I could define 17 as a constant, 29 as a constant etc. ?

Comment (livecodeali) [inline extensions/widgets/pushButton/pushButton.lcb]: Yeah probably have separate numerator and denominator constants is a good plan. If we don't already have a bug report / feature request about evaluated constants then can you file one?

Comment (livecodegeorgia) [inline extensions/widgets/pushButton/pushButton.lcb]: Will do!

**Comment by livecodeali** [comment]
One (i think final) thing is that you should remove the existing button
git rm -r extensions/button

and modify the extensions.gyp file so that this button gets built in the extensions project instead of the old one.

**Comment by livecodeali** [comment]
@livecode-vulcan review ok 2d71a56

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 2d71a5683e2b77375034cd39b133d284fc73c453

**Comment by livecode-vulcan** [comment]
😎 test success 2d71a5683e2b77375034cd39b133d284fc73c453
- try-community-linux-x86: [success](https://vulcan.livecode.com/builders/try-community-linux-x86/builds/402)
- try-community-mac: [success](https://vulcan.livecode.com/builders/try-community-mac/builds/402)
- try-community-linux-x86_64: [success](https://vulcan.livecode.com/builders/try-community-linux-x86_64/builds/401)
- try-community-emscripten: [success](https://vulcan.livecode.com/builders/try-community-emscripten/builds/400)
- try-community-ios: [success](https://vulcan.livecode.com/builders/try-community-ios/builds/406)
- try-community-win-x86: [success](https://vulcan.livecode.com/builders/try-community-win-x86/builds/397)
- try-community-android-armv6: [success](https://vulcan.livecode.com/builders/try-community-android-armv6/builds/408)

**Comment by livecodeali** [comment]
@livecode-vulcan review ok  2a2e984

**Comment by livecode-vulcan** [comment]
💙 review by @livecodeali ok 2a2e984a9cde70ed5b0a06f070570a055c329c50

**Comment by livecode-vulcan** [comment]
⁉ merge unchanged 2a2e984a9cde70ed5b0a06f070570a055c329c50 → 2734b3f52fcebaff6f01eea8cb236fd2b44938ed

Pull request is already fully merged!

**Pull request merged**